kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableReadsSuite.scala

```diff
@@ -571,6 +571,19 @@ class DeltaTableReadsSuite extends AnyFunSuite with TestUtils {
       expectedAnswer = (0 until 150).map(i => TestRow(i.toLong)))
   }
 
+  test("table with checkpoint and log compaction") {
```

Comment:
should we do something to test that we are in fact reading the compaction files for this? I think there's no way to know how we are loading this data right?

I think we already have utils for this that can save what files we read I can help find them if needed

Reply:
also as discussed we should (1) test with the compactions written by Kernel (2) test with compaction files with all the possible actions in it (removes, domains metadatas, etc)

Reply:
This is now tested. I only tested adds/removes. The `LogCompactionWriterSuite` verifies that all supported actions are preserved, and we read these compactions as "normal" json files, so I believe all actual functionality is tested. I'd be quite a lot more test code to check all these actions, so I've opted not to for now. If you think it's important, let me know why, and I can look at adding them in.

Reply:
Why is it a lot more code to also check the other actions? 

For example for DM, I think you can just set a domain in one of the commits and then to check it just use `snapshot.getDomain` in the check?

And then for like a metadata action, just change a table property?

Reply:
I would be more inclined to prefer 1 test that has a compaction file with everything (all actions) in it and checks everything but it doesn't need to be repeated with all the different cases you've added below. I think those are fairly well covered by the LogSegmentSuite